crates/bevy_transform/src/lib.rs

```diff
@@ -105,21 +105,22 @@ impl Plugin for TransformPlugin {
                 schedule.configure_set(
                     TransformSystem::TransformPropagate.in_base_set(StartupSet::PostStartup),
                 );
+                propagate_in_schedule(schedule);
             })
-            // FIXME: https://github.com/bevyengine/bevy/issues/4381
-            // These systems cannot access the same entities,
-            // due to subtle query filtering that is not yet correctly computed in the ambiguity detector
-            .add_startup_system(
-                sync_simple_transforms
-                    .in_set(TransformSystem::TransformPropagate)
-                    .ambiguous_with(PropagateTransformsSet),
-            )
-            .add_startup_system(propagate_transforms.in_set(PropagateTransformsSet))
-            .add_system(
-                sync_simple_transforms
-                    .in_set(TransformSystem::TransformPropagate)
-                    .ambiguous_with(PropagateTransformsSet),
-            )
-            .add_system(propagate_transforms.in_set(PropagateTransformsSet));
+            .edit_schedule(CoreSchedule::FixedUpdate, propagate_in_schedule)
+            .edit_schedule(CoreSchedule::Main, propagate_in_schedule);
     }
 }
+
+fn propagate_in_schedule(schedule: &mut Schedule) {
+    // FIXME: https://github.com/bevyengine/bevy/issues/4381
+    // These systems cannot access the same entities,
+    // due to subtle query filtering that is not yet correctly computed in the ambiguity detector
+    schedule
+        .add_system(
+            sync_simple_transforms
+                .in_set(TransformSystem::TransformPropagate)
+                .ambiguous_with(PropagateTransformsSet),
+        )
+        .add_system(propagate_transforms.in_set(PropagateTransformsSet));
+}
```

Comment:
Instead of adding systems to an `&mut Schedule`, can you make this return a `SystemConfigs` struct? Then the caller would just call `add_systems` with the result to add both systems to a schedule.